v3/src/data-model/data-set.ts

```diff
@@ -268,6 +268,18 @@ export const DataSet = types.model("DataSet", {
     }
   }
 
+  function getUniqueAttributeName(baseName: string, allowNames: (string | null)[]) {
```

Comment:
There is already a `uniqueName()` function in `js-utils` that I think could be used to accomplish this. You'd just need to pass an `isValid` function that looked something like:
```
(name: string) => (name === existingName) || !self.attributes.find(attr => name === attr.name)
```